src/main/java/seedu/address/logic/commands/NewtagCommand.java

```diff
@@ -26,6 +27,13 @@ public NewtagCommand(Tag tag) {
 
     @Override
     public CommandResult execute(Model model) {
-        return new CommandResult("Valid newtag Command Received: " + tag);
+        requireAllNonNull(model);
+        boolean success = model.addTag(tag);
+        if (!success) {
+            return new CommandResult(MESSAGE_DUPLICATE);
+        }
+        String successMsg = MESSAGE_SUCCESS + " " + tag + "\n";
```

Comment:
I think successMessage would be a better name

Reply:
I agree, thanks!